### PR TITLE
MAIN - adds a property on modal that will disable/enable submit button on modal

### PR DESCRIPTION
**Pull request by paschalidi**
hello you 👋 👋 @VivyTeam/web 

There are some use cases have submission disabled in our `Modal` components.

This PR will add a property `submissionEnabled ` on the Modal that will make this possible.

### CODE REVIEW

**Comment by paschalidi** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>✅  Get a 🏆 for adding description on this PR. - <i>Celebrate on every step!</i></td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>✅ Get a 👍 for requesting reviewers. - <i>Celebrating every win!</i></td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against eae88e028058e594d1cbce84aec6c9363b31c11c
</p>

File: src/Modal/Modal.jsx [hunk 1]
```diff
@@ -149,6 +154,7 @@ Modal.defaultProps = {
   cancelText: "",
   submitText: "",
   role: "dialog",
+  submissionEnabled: false,
```
Comment (Weetbix) [inline src/Modal/Modal.jsx]: think a more sensible default might be true, seems like the most common use case (info prompts etc)

File: stories/Modal.story.jsx [hunk 2]
```diff
@@ -147,4 +150,44 @@ storiesOf("Modal", module)
     {
       info: "Scrolling should be disabled when modal is open"
     }
-  );
+  )
+  .add("toggle availability of submit button", () => {
+    const ModalUseCase = () => {
+      const [open, toggleModal] = useState(true);
+      const [actionComplete, toggleSubmissionState] = useState(false);
```
Comment (Weetbix) [inline stories/Modal.story.jsx]: `actionComplete` might be better named `allowSubmit`?

**Review by Weetbix — CHANGES_REQUESTED** [review]
Cool 😎 Few minor comments

**Comment by paschalidi** [comment]
@Weetbix the changes you requested are there by now! Thanks for reviewing!

**Review by Weetbix — APPROVED** [review]
Thank you sir!

**Pull request merged**